### PR TITLE
Catalog provider filter

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Use the new design in the provider selection filter in the catalog view. Some additional storybook stories added.
I didn't work on the whole catalog story as other work to add the accordion to the catalog is on progress, and it will impact completely this view.

**I'm sending the PR to `new-checks-ui` to not disturb the "almost ready realease". We can change if you want anyway.**

gif
gif2

## How was this tested?

Old tests are enough as the functionality doesn't change.

### CODE REVIEW

**Review by jagabomb — APPROVED** [review]
LGTM

File: assets/js/components/ChecksCatalog/ProviderSelection.stories.jsx
Comment (EMaksy) [inline assets/js/components/ChecksCatalog/ProviderSelection.stories.jsx:1]: Just a question, could we not import the providerData from ProviderLabel and get the data from the source, so if a new provider label is added, the story has a complete list?

Comment (arbulu89) [reply assets/js/components/ChecksCatalog/ProviderSelection.stories.jsx:1]: Of course we could.
The purpose of this story is not to show all the providers, it is to show how the component works.
Anyway, I will grant your wish XD

**Review by EMaksy — APPROVED** [review]
LGTM :D

File: assets/js/components/ChecksCatalog/ChecksCatalog.jsx [hunk 1]
```diff
@@ -35,34 +37,27 @@ function ChecksCatalog() {
   } = useSelector(getCatalog());
 
   useEffect(() => {
-    const apiParams =
-      selectedProvider === ALL_FILTER
-        ? {}
-        : { provider: getProviderByLabel(updatedProvider, selectedProvider) };
-
-    dispatch(updateCatalog(apiParams));
+    dispatch(updateCatalog(buildUpdateCatalogPayload(selectedProvider)));
```
Comment (nelsonkopliku) [inline assets/js/components/ChecksCatalog/ChecksCatalog.jsx:40]: What about reducing function call nesting by having a `buildUpdateCatalogAction` function?

```js
dispatch(buildUpdateCatalogAction(selectedProvider));
```

File: assets/js/components/ChecksCatalog/ChecksCatalog.jsx [hunk 2]
```diff
@@ -8,20 +8,22 @@ import { getCatalog } from '@state/selectors/catalog';
...
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
 import ProviderSelection from './ProviderSelection';
 
-const ALL_FILTER = 'All';
+const ALL_FILTER = 'all';
+const ALL_FILTER_TEXT = 'All';
 const updatedProvider = {
-  default: { label: ALL_FILTER },
+  [ALL_FILTER]: { label: ALL_FILTER_TEXT },
```
Comment (nelsonkopliku) [inline assets/js/components/ChecksCatalog/ChecksCatalog.jsx:21]: This extended providerData (which also contains the `all`) could also make sense to be close to where we define the available providers in `@components/ProviderLabel/ProviderLabel` or even in `ProviderSelection`.

Not a biggie though.

**Review by nelsonkopliku — APPROVED** [review]
LGTM. I just noted a couple of things that came to my mind, which are not blocking tho.

I will change my work on the accordion to be against `new-checks-ui` branch

File: assets/js/components/ChecksCatalog/ChecksCatalog.jsx (hunk 1 above)
Comment (arbulu89) [reply assets/js/components/ChecksCatalog/ChecksCatalog.jsx]: Change

File: assets/js/components/ChecksCatalog/ChecksCatalog.jsx (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/components/ChecksCatalog/ChecksCatalog.jsx:21]: Alright, let's keep it as it is :+1:

**Pull request merged**